fusequery/query/src/optimizers/optimizer_projection_push_down_test.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020-2021 The Datafuse Authors.
+//
+// SPDX-License-Identifier: Apache-2.0.
+
+#[test]
```

Comment:
It would be better to mock a logic plan like:  select a from t where a>10 and    b<6

Reply:
If I understand correctly, this statement needs three plans: projection, filter, and readdatasource, correct?

Reply:
> If I understand correctly, this statement needs three plans: projection, filter, and readdatasource, correct?

It can help us to check the push down logic:
select a from tb1 where a>10 and b <6
push down: [a, b]

Reply:
I have changed the second unit test.
https://github.com/datafuselabs/datafuse/pull/286/files#diff-8713bb30d19e04fd3b030f73f92d5126560d536bd287f36ba2f3de281873bb54R91

Is the test correct?

Reply:
I think in the optimizer code I also need to handle FilterPlan explicitly. I will fix it.

Reply:
@zhang2014 @BohuTANG  I fixed test_projection_push_down_optimizer_2. The code can push down `a `and `b`
for statement `select a from t where a > 6 and b <= 10` now.

Reply:
`ORDER BY` and `GROUP BY` also need to be tested. maybe a better test example:
```
SELECT a FROM table WHERE b = 10 GROUP BY a, C HAVING d < 9 ORDER BY e LIMIT 10;
```

Reply:
Sure. This test seems very comprehensive.

Reply:
@zhang2014  @BohuTANG  How should I create a plan for this SQL statement? I tried build_from_sql. It seems build_from_sql can only use numbers_mt table.

Reply:
We do'nt need the numbers_mt table.
We can assemble any sql statement with the plan builder, a  simple builder is here:
https://github.com/datafuselabs/datafuse/blob/master/common/planners/src/plan_aggregator_test.rs#L14

Here difference is we have a data source schema.

Reply:
What is the expected output for this statement after pushdown optimization?
Is this new unit test what we want?

https://github.com/datafuselabs/datafuse/pull/286/files#diff-2a4cdcaf99f7e3b057e557b51f558c00905dde68b61689580c5d1f3e49e30e61R421

Reply:
What is the expected output for this statement after pushdown optimization?
Is this new unit test what we want?

https://github.com/datafuselabs/datafuse/pull/286/files#diff-2a4cdcaf99f7e3b057e557b51f558c00905dde68b61689580c5d1f3e49e30e61R421

Reply:
@zhang2014 @BohuTANG  Does the test below meet the requirement? 

https://github.com/datafuselabs/datafuse/pull/286/files#diff-2a4cdcaf99f7e3b057e557b51f558c00905dde68b61689580c5d1f3e49e30e61R433